src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -26,24 +28,44 @@
     private long threadLocalSetupTimeDiff = 0;
     private long pairHMMSetupTime = 0;
 
+    public enum Implementation {AVX, OMP}
+
     private final PairHMMNativeBinding pairHmm;
 
     //Hold the mapping between haplotype and index in the list of Haplotypes passed to initialize
     //Use this mapping in computeLikelihoods to find the likelihood value corresponding to a given Haplotype
     private final Map<Haplotype, Integer> haplotypeToHaplotypeListIdxMap = new LinkedHashMap<>();
     private HaplotypeDataHolder[] mHaplotypeDataArray;
 
-    public VectorLoglessPairHMM(PairHMMNativeArguments args) throws UserException.HardwareFeatureException {
-        // TODO: connect GATK temp directory
-        final boolean isSupported = new IntelPairHmm().load(null);
-
-        if (!isSupported) {
-            throw new UserException.HardwareFeatureException("Machine does not support AVX PairHMM.");
+    public VectorLoglessPairHMM(Implementation implementation,
+                                PairHMMNativeArguments pairHmmArgs) throws UserException.HardwareFeatureException {
+
+        // require non-null implementation and pairHmmArgs
+        Utils.nonNull(implementation);
+        Utils.nonNull(pairHmmArgs);
+
+        switch (implementation) {
+            case AVX:
+                final boolean isAvxSupported = new IntelPairHmm().load();
+                if (!isAvxSupported) {
+                    throw new UserException.HardwareFeatureException("Machine does not support AVX PairHMM.");
+                }
+                pairHmm = new IntelPairHmm();
+                break;
+
+            case OMP:
+                final boolean isOmpSupported = new IntelPairHmmOMP().load();
+                if (!isOmpSupported) {
+                    throw new UserException.HardwareFeatureException("Machine does not support OpenMP AVX PairHMM.");
+                }
+                pairHmm = new IntelPairHmmOMP();
+                break;
+
+            default:
+                throw new UserException.HardwareFeatureException("Unknown PairHMM implementation.");
```

Comment:
This should probably be a `GATKException` since it's a bug in the gatk rather than a hardware issue.  Also, could you include the implementation in the error message?